ctk/common/src/main/kotlin/org/codice/compliance/verification/core/SamlExtensionsVerifier.kt

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compliance.verification.core
+
+import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.SAMLCore_SamlExtensions
+import org.codice.compliance.attributeList
+import org.codice.compliance.children
+import org.codice.compliance.recursiveChildren
+import org.codice.compliance.utils.TestCommon
+import org.w3c.dom.Attr
+import org.w3c.dom.Element
+import org.w3c.dom.Node
+
+class SamlExtensionsVerifier(val node: Node) {
```

Comment:
❓ Wouldn't schema validation be sufficient to affirm these extensions? At the least, it appears that `SubjectConfirmationData` is covered by the schema. Unless `lax` processing is laxer than I think, this snippet should ensure that no extended attributes or elements are added in the SAML namespace:
```xml
<element name="SubjectConfirmationData"
         type="saml:SubjectConfirmationDataType"/>
<complexType name="SubjectConfirmationDataType" mixed="true">
<complexContent>
    <restriction base="anyType">
        <sequence>
            <any namespace="##any" processContents="lax" minOccurs="0"
                 maxOccurs="unbounded"/>
        </sequence>
        <attribute name="NotBefore" type="dateTime" use="optional"/>
        <attribute name="NotOnOrAfter" type="dateTime" use="optional"/>
        <attribute name="Recipient" type="anyURI" use="optional"/>
        <attribute name="InResponseTo" type="NCName" use="optional"/>
        <attribute name="Address" type="string" use="optional"/>
        <anyAttribute namespace="##other" processContents="lax"/>
    </restriction>
</complexContent>
</complexType>
```

Reply:
I will check on that.